pkg/sql/schemachanger/scplan/plan.go

```diff
@@ -108,14 +108,35 @@ func MakePlan(ctx context.Context, initial scpb.CurrentState, params Params) (p
 func makePlan(ctx context.Context, p *Plan) (err error) {
 	{
 		start := timeutil.Now()
+		var oldCurrent []scpb.Status
+		if p.Params.ExecutionPhase <= scop.PreCommitPhase {
+			// We need the full graph at this point because the plan may transition
+			// back to the initial statuses at pre-commit time.
+			oldCurrent = append(oldCurrent, p.CurrentState.Current...)
+			for i, t := range p.Targets {
+				p.CurrentState.Current[i] = scpb.AsTargetStatus(t.TargetStatus).InitialStatus()
+			}
+		}
 		p.Graph = buildGraph(ctx, p.Params.ActiveVersion, p.CurrentState)
+		// Restore the original current statuses.
+		if oldCurrent != nil {
+			copy(p.CurrentState.Current, oldCurrent)
```

Comment:
why not just `p.CurrentState.Current = oldCurrent`? You already allocated the thing and go gc is not generational, nor do I expect it to become so in the next half-decade.

Reply:
The idea was that the `scpb.CurrentState` embedded in the `Plan` does not belong to the `Plan` so I want to leave it the same way I found it after swapping out the `Current` slice for the purpose of generating the graph.

Reply:
In any case my code was bad and I'm replacing it with something better.